charmtools/build/tactics.py

```diff
@@ -978,18 +984,15 @@ def trigger(cls, entity, target, layer, next_config):
         return True
 
     def _try_to_get_current_sha(self):
-        cmds = (
-            ('git', 'describe', '--dirty', '--always'),
-            ('bzr', 'version-info'),
-            ('hg', 'id', '-n'),
-        )
         with utils.cd(str(self.charm)):
-            for cmd in cmds:
+            for cmd in self.CMDS:
                 try:
                     sha = utils.Process(cmd)()
                     if sha:
                         return sha.output
-                except FileNotFoundError as e:
+                except OSError as e:
+                    if e.errno != errno.ENOENT:
+                        raise
```

Comment:
So if the git command fails for any reason other than not being installed on the system, then an exception will be raised and it won't proceed to do the bzr or hg check? It seems perfectly normal for git commands to fail if run in a bzr or hg branch.

Reply:
If the git command can be run but exits with a non-zero return code, it won't raise an exception and will instead set `sha.exit_code`.  I've updated the PR to explicitly check the exit code and also added some more debug logging.